src/libraries/Native/Unix/System.Native/pal_process.c

```diff
@@ -293,9 +325,9 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
 
     // Open pipes for any requests to redirect stdin/stdout/stderr and set the
     // close-on-exec flag to the pipe file descriptors.
-    if ((redirectStdin  && SystemNative_Pipe(stdinFds,  PAL_O_CLOEXEC) != 0) ||
-        (redirectStdout && SystemNative_Pipe(stdoutFds, PAL_O_CLOEXEC) != 0) ||
-        (redirectStderr && SystemNative_Pipe(stderrFds, PAL_O_CLOEXEC) != 0))
+    if ((redirectStdin  && SocketPair(stdinFds) != 0) ||
+        (redirectStdout && SocketPair(stdoutFds) != 0) ||
+        (redirectStderr && SocketPair(stderrFds) != 0))
     {
         success = false;
         goto done;
```

Comment:
Unrelated to this patch (and GitHub doesn't let me comment there, grr), but there's no error recovery for the `pipe2()` call below.

Reply:
That's called out in the comment:
https://github.com/dotnet/runtime/pull/34861/files/d102eb5d33272cb95e43c16909ddce0903077433#diff-df5b2051e73600a29fbe0b3230daf7d1R340